app/controllers/administrative-units/administrative-unit/sites/site/edit.js

```diff
@@ -0,0 +1,76 @@
+import Controller from '@ember/controller';
+import { dropTask } from 'ember-concurrency';
+import { inject as service } from '@ember/service';
+import { tracked } from '@glimmer/tracking';
+
+export default class AdministrativeUnitsAdministrativeUnitSitesSiteEditController extends Controller {
+  @service router;
+  @tracked isPrimarySite =
+    this.site.get('id') === this.administrativeUnit.get('primarySite.id');
+
+  get site() {
+    return this.model.adminUnitSite.site;
+  }
+
+  get administrativeUnit() {
+    return this.model.adminUnitSite.administrativeUnit;
+  }
+
+  @dropTask
+  *save(event) {
+    event.preventDefault();
+
+    let address = yield this.site.address;
+
+    if (address.hasDirtyAttributes) {
```

Comment:
Can you add a TODO comment here? I think we should make this reusable since we'll need it on multiple edit pages. I still need to think of a good way to tackle that though :smile: